addons/pos_self_order/static/src/app/services/self_order_service.js

```diff
@@ -773,19 +782,43 @@ export class SelfOrder extends Reactive {
                     line.customer_note = alreadySent.customer_note;
                     line.selected_attributes = alreadySent.selected_attributes;
                 } else {
+                    const productName =
+                        wrongChild?.product_id?.name &&
+                        !line.product_id?.self_order_available &&
+                        line.product_id?.name
+                            ? line.product_id.name
+                            : wrongChild?.product_id?.name || line.product_id?.name;
```

Comment:
Couldn't this be resume like
```suggestion
                    const productName = line.product_id.self_order_available ? line.product_id.name : wrongChild.product_id.name;
```